presto-orc/src/main/java/com/facebook/presto/orc/writer/StreamOrderingLayout.java

```diff
@@ -0,0 +1,212 @@
+    {
+        private final int column;
+        private final DwrfProto.KeyInfo key;
+
+        public ColumnKeyInfo(int column, DwrfProto.KeyInfo key)
+        {
+            this.column = column;
+            this.key = requireNonNull(key, "inputKeyId cannot be null");
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == null) {
+                return false;
+            }
+            if (!(obj instanceof ColumnKeyInfo)) {
+                return false;
+            }
+            ColumnKeyInfo input = (ColumnKeyInfo) obj;
+            return this.column == input.column && this.key.equals(input.key);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(column, key);
+        }
+    }
+
+    private static class ColumnSequenceInfo
+    {
+        private final int column;
+        private final int sequence;
+
+        public ColumnSequenceInfo(int column, int sequence)
+        {
+            this.column = column;
+            this.sequence = sequence;
+        }
+
+        @Override
+        public boolean equals(Object obj)
+        {
+            if (obj == null) {
+                return false;
+            }
+            if (!(obj instanceof ColumnSequenceInfo)) {
+                return false;
+            }
+            ColumnSequenceInfo input = (ColumnSequenceInfo) obj;
+            return this.column == input.column && this.sequence == input.sequence;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(column, sequence);
+        }
+    }
+
+    private StreamMetadata getStreamMetadata(
+            Map<Integer, Integer> nodeIdToColumn,
+            Map<Integer, ColumnEncoding> nodeIdToColumnEncodings,
+            DwrfStreamOrderingConfig config)
+    {
+        ImmutableMap.Builder<ColumnKeyInfo, Integer> keyToSequenceBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<ColumnSequenceInfo, List<StreamDataOutput>> sequenceToStreamsBuilder = ImmutableMap.builder();
+        Map<Integer, Set<DwrfProto.KeyInfo>> columnToKeySet = config.getColumnToKeySet();
+
+        // iterate through all the encodings and if the encoding has additional sequence encodings put it in the map
+        for (Map.Entry<Integer, ColumnEncoding> entry : nodeIdToColumnEncodings.entrySet()) {
+            int nodeId = entry.getKey();
+            int column = nodeIdToColumn.get(nodeId);
+
+            if (entry.getValue().getAdditionalSequenceEncodings().isPresent() && columnToKeySet.containsKey(column)) {
+                // add entries only if stream ordering contains the column ID
+                Set<DwrfProto.KeyInfo> keysPerColumn = columnToKeySet.get(column);
+                for (Map.Entry<Integer, DwrfSequenceEncoding> sequenceToEncoding : entry.getValue().getAdditionalSequenceEncodings().get().entrySet()) {
+                    int sequence = sequenceToEncoding.getKey();
```

Comment:
Use Integer instead of int to avoid extra boxing in `keyToSequenceBuilder.put(new ColumnKeyInfo(column, key), sequence)`.